Issues with Cmake after commit "[SofaGeneralObjectInteraction] BoxStiffSpringForceField is now a pack…"

### Discussed in https://github.com/sofa-framework/sofa/discussions/2657

<div type='discussions-op-text'>

<sup>Originally posted by **longhathuc** February  2, 2022</sup>
Hi, I'm Long from mimesis. I have tried to build the plugins with recent commit of sofa at https://github.com/mimesis-inria/sofa/ But failed.

Sofa build successfully, but for the plugins, this is the error when doing cmake

 ```
   CMake Error at /home/long/work/install/sofaInstall/lib/cmake/SofaGeneral/SofaGeneralConfig.cmake:40 (find_package):
    Found package configuration file:

    1 /home/long/work/install/sofaInstall/plugins/SofaGeneralObjectInteraction/lib/cmake/SofaGeneralObjectInteraction/SofaGeneralObjectInteractionConfig.cmake

    but it set SofaGeneralObjectInteraction_FOUND to FALSE so package
    "SofaGeneralObjectInteraction" is considered to be NOT FOUND. Reason given
    by package:

    The following imported targets are referenced, but are missing: SofaEngine

    Call Stack (most recent call first):
    /home/long/work/sperry/CMakeLists.txt:56 (find_package)
```

Do you have any clue which might cause the error?

Thanks </div>

#2621 adds two new dependencies on SofaGeneralEngine and SofaEngine for SofaGeneralObjectInteraction, but its associated cmake.in did not reflect the change.

--- a/modules/SofaGeneralObjectInteraction/SofaGeneralObjectInteractionConfig.cmake.in
+++ b/modules/SofaGeneralObjectInteraction/SofaGeneralObjectInteractionConfig.cmake.in
@@ -3,7 +3,9 @@
 @PACKAGE_GUARD@
 @PACKAGE_INIT@
 
-find_package(SofaDeformable QUIET REQUIRED) # SofaDeformable
+find_package(SofaDeformable QUIET REQUIRED)
+find_package(SofaEngine QUIET REQUIRED)
+find_package(SofaGeneralEngine QUIET REQUIRED)
 
 if(NOT TARGET @PROJECT_NAME@)
     include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")

